src/App.tsx

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import style from './App.module.scss';
+import Header from './components/Header/Header';
+import CardItems from './components/CardItems/CardItems';
+import { URL } from './models/enums';
+import { ShowData } from './models/interfaces';
+import { MdReportGmailerrorred } from 'react-icons/md';
+import ErrorBoundary from './components/ErrorBoundary/ErrorBoundary';
+
+interface AppProps {
+  shows: ShowData[];
+  isLoading: boolean;
+  error: Error | null | unknown;
+  currentPage: number;
+  itemsPerPage: number;
+  isMoreShows: boolean;
+  searchQuery: string;
+  isShowMoreButtonDisable: boolean;
+  isClickedErrorButton: boolean;
+}
+class App extends React.Component<object, AppProps> {
+  constructor(props: AppProps) {
+    super(props);
+    this.state = {
+      shows: [],
+      isLoading: false,
+      error: null,
+      currentPage: 0,
+      itemsPerPage: 20,
+      isMoreShows: false,
+      searchQuery: '',
+      isShowMoreButtonDisable: false,
+      isClickedErrorButton: false,
+    };
+
+    this.loadShows = this.loadShows.bind(this);
+    this.showMoreShows = this.showMoreShows.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+    this.enableButton = this.enableButton.bind(this);
+    this.throwError = this.throwError.bind(this);
+  }
+
+  async componentDidMount() {
```

Comment:
Should it be async if you don't use await inside?

Reply:
I fixed it for functional components